src/actions/search-results.js

```diff
@@ -278,59 +115,42 @@ function constructDepartureAirportQuery (departureAirport) {
 
 export function startSearch () {
   return (dispatch, getState) => {
-    const {
-       search: {
-        tags
-        },
-       travelInfo: {
-        numberOfChildren,
-        numberOfAdults,
-        childAge1,
-        childAge2,
-        childAge3,
-        childAge4,
-        departureDate,
-        duration,
-        departureAirport
-      }
-    } = getState();
-    const childAgeArray = [childAge1, childAge2, childAge3, childAge4];
+    const store = getState();
+    const { search: { tags, fingerprint: clientId, socketConnectionId: connectionId } } = store;
     if (tags.length > 0) {
       dispatch(busySearching(true));
-      const formattedTags = formatQuery(tags);
-      const passengers = combinePassengersForQuery(childAgeArray, numberOfChildren, numberOfAdults);
-      const departureAirports = constructDepartureAirportQuery(departureAirport);
-      const travelPeriod = constructTravelPeriodQuery(departureDate, duration);
-      const query = {passengers: passengers, travelPeriod: travelPeriod, departureAirports: departureAirports, ...formattedTags};
+      const query = formatQuery(store);
       console.log('query', JSON.stringify(query));
       return graphqlService
-        .query(MUTATION_START_SEARCH, {'query': JSON.stringify(query)})
+        .query(MUTATION_START_SEARCH, {'query': JSON.stringify(query), clientId, connectionId})
```

Comment:
we *also* need to know the `userId` of the client. this is the `id` that is saved on the device and will be re-used if the person has multiple sessions.